app/components/form/FullPageForm.tsx

```diff
@@ -81,7 +81,7 @@ export function FullPageForm<TFieldValues extends FieldValues>({
       <form
         className="ox-form pb-20"
         id={id}
-        onSubmit={(e) => {
+        onSubmit={async (e) => {
```

Comment:
My understanding is if there are no `await`s in the function, this shouldn't do anything. Does it make a difference for you?

Reply:
Hmm, will it handle the `await readBlobAsBase64` without that?

Reply:
Pretty sure it is ok. @charliepark @zephraph let me know if you think any of this is not correct.

This is a kind of evil thing about `async/await` in JS, though it's sort of a nice thing: all JS code is already asynchronous, basically. You can call `setTimeout` anywhere, execute a function that returns a promise anywhere. All `async` on a function means is that you might call `await` directly inside that function, directly in the sense of literally right there in the definition of the function. It's primarily a syntactic thing.

The instance-create `onSubmit`, where the `await readBlobAsBase64` takes place, _is_ marked `async`, necessarily so because you are using `await` in there. The tricky thing to understand is that the `await` aspect of that is only about sequencing events _within_ that function. What the `await` is primarily doing there is ensuring that `createInstance.mutate()` does not happen until after `readBlobAsBase64` is done. But that is about sequencing inside the function. Because you are not calling `await` inside of `FullPageForm` `onSubmit` (plus `handleSubmit` is the last thing in the function anyway), there is no sequencing question. This function ends and whatever async ops are put on the queue are just going to do whatever they do — we don't don't wait for them to finish, and whatever calls this function also doesn't wait.

Now... `form.handleSubmit` does return a promise, so you actually could `await` on line 94, and then you would have to keep the `async`. But I'm not sure whether it accomplishes anything. In order for it to matter, there would have to be something waiting for this `onSubmit` to finish, but since this is being passed to native `<form>`, I doubt it. RHF is not doing anything like that. It's all regular props and function calls.